app/routes/calendar.ts

```diff
@@ -1,87 +1,148 @@
 import "dotenv/config";
 import ical from "ical.js";
+import { getConfig } from "~/config.server";
 
 export async function loader() {
 	console.log("hit!");
 	const now = new Date();
 
-	const req = await fetch(process.env.CAL_URL!);
-	if (!req.ok) throw await req.text();
-	const icsText = await req.text();
+	const config = await getConfig();
+	const calendarUrl = config.calendarUrl || process.env.CAL_URL || "";
 
-	const jcalData = ical.parse(icsText);
-	const vcalendar = new ical.Component(jcalData);
-
-	// const vtzs = vcalendar.getAllSubcomponents("vtimezone");
-	// vtzs.forEach((vtz) => {
-	//   const tz = new ical.Timezone(vtz);
-	//   ical.TimezoneService.register(tz);
-	// });
+	if (!calendarUrl) {
+		return new Response(
+			JSON.stringify({
+				currentEvent: null,
+				upcomingEvents: [],
+			}),
+			{
+				status: 200,
+				headers: {
+					"Content-Type": "application/json",
+				},
+			},
+		);
+	}
 
-	const vevents = vcalendar.getAllSubcomponents("vevent");
-	const thisday = vevents
-		.map((e) => {
-			const event = new ical.Event(e);
-			// event.startDate = event.startDate.convertToZone(
-			//   ical.TimezoneService.get("Europe/Athens")
-			// );
-			// event.endDate = event.endDate.convertToZone(
-			//   ical.TimezoneService.get("Europe/Athens")
-			// );
-			return event;
-		})
-		.filter((event) => {
-			const dayEnd = new Date(
-				now.getFullYear(),
-				now.getMonth(),
-				now.getDate(),
-				23,
-				59,
-				59,
-				999,
-			);
-			return (
-				(event.endDate.toJSDate() <= dayEnd &&
-					event.startDate.toJSDate() >= now) ||
-				(event.startDate.toJSDate() <= now &&
-					event.endDate.toJSDate() >= now)
+	try {
+		const req = await fetch(calendarUrl);
+		if (!req.ok) {
+			console.error("Failed to fetch calendar:", await req.text());
+			return new Response(
```

Comment:
please return an meaningful error message here somehow so that we can inform the user that something went wrong while fetching the ical